homeassistant/components/cast/config_flow.py

```diff
@@ -28,7 +33,22 @@ async def async_step_import(self, import_data=None):
         """Import data."""
         if self._async_current_entries():
             return self.async_abort(reason="single_instance_allowed")
-        data = {CONF_KNOWN_HOSTS: self._known_hosts}
+
+        media_player_config = self.hass.data[DOMAIN].get("media_player") or {}
+        if not isinstance(media_player_config, list):
+            media_player_config = [media_player_config]
+        for cfg in media_player_config:
+            try:
+                cfg = ENTITY_SCHEMA(cfg)
+                if CONF_IGNORE_CEC in cfg:
+                    self._ignore_cec.extend(cfg[CONF_IGNORE_CEC])
+                if CONF_UUID in cfg:
+                    self._wanted_uuid.append(cfg[CONF_UUID])
+            except vol.Error:
+                # Invalid config, ignore it
+                pass
```

Comment:
It would be better to validate the configuration yaml in the `CONFIG_SCHEMA` in the init module and give feedback to users if there's invalid config.

Reply:
Yes, that would be better, but the point of this PR is to deprecated yaml config so I don't think it adds much value?

Reply:
As long as we do import the configuration yaml, we should do it according to best practices.

Reply:
Alright, improved in fa14da8